mplc/multi_partner_learning.py

```diff
@@ -603,12 +605,65 @@ def fit_minibatch(self):
         logger.debug("End of S-Model collaborative round.")
 
 
-# Supported multi-partner learning approaches
+class EnsemblePredictions(MultiPartnerLearning):
+    """
+    Ensemble (average) prediction of several input models
+    This approach can only be used with the EnsemblePredictionsModel
+    """
+
+    def __init__(self, scenario, **kwargs):
+        # First, if only one partner, fall back to dedicated single partner function
+        super(EnsemblePredictions, self).__init__(scenario, **kwargs)
+
+        if self.partners_count == 1:
+            raise ValueError('Only one partner is provided. Please use the dedicated SinglePartnerLearning class')
+
+        partner_model_list = [self.dataset.generate_new_model() for _ in range(self.partners_count)]
+        self.model = EnsemblePredictionsModel(partner_model_list)
+
+        for partner in self.partners_list:
+            partner.model_weights = self.model_weights
+        # the partners should all start with a different set of weights
```

Comment:
Hmmm, how does this comment combines with the code around it?

Reply:
Ah yes a TODO that was not done! I should modify that.